modules/javafx.graphics/src/main/java/javafx/css/StyleableLongProperty.java

```diff
@@ -116,21 +118,26 @@ public void set(long v) {
     @Override
     public StyleOrigin getStyleOrigin() { return origin; }
 
-    private StyleOrigin origin = null;
-    private TransitionMediatorImpl mediator = null;
+    private StyleOrigin origin;
+    private TransitionMediatorImpl mediator;
 
     private final class TransitionMediatorImpl extends TransitionMediator {
-        private final long oldValue;
-        private final long newValue;
-
-        public TransitionMediatorImpl(long oldValue, long newValue) {
-            this.oldValue = oldValue;
-            this.newValue = newValue;
+        private final long startValue;
+        private final long endValue;
+        private long reversingAdjustedStartValue;
+
+        public TransitionMediatorImpl(long startValue, long endValue) {
+            this.startValue = startValue;
+            this.endValue = endValue;
+            this.reversingAdjustedStartValue = startValue;
         }
 
         @Override
         public void onUpdate(double progress) {
-            set(progress < 1 ? oldValue + (long)((newValue - oldValue) * progress) : newValue);
+            // Longs are interpolated in real number space and rounded to the nearest long.
+            long diff = endValue - startValue;
+            long result = startValue + Math.round(progress * diff);
+            set(progress < 1 ? Utils.clamp(startValue, result, endValue) : endValue);
```

Comment:
`clamp` won't solve that if `progress = 1.0` that the calculation may result in a value less than `endValue`.  You need both the `clamp` and a special case:

    set(progress == 1.0 ? endValue : progress < 1 ? Utils.clamp(startValue, result, endValue) : endValue);

Use `==` or `>=` I don't know what's better.

Reply:
I don't understand this. If progress != 1, then it must be < 1 by definition. It can never be > 1.
So the current implementation will just ignore the result of the computation for the special case (result = 1):
```java
set(progress < 1 ? Utils.clamp(startValue, result, endValue) : endValue);
```

Reply:
Sorry, you're right. I completely missed it.